plugins/woocommerce-blocks/assets/js/blocks/product-collection/utils.tsx

```diff
@@ -166,24 +167,63 @@ export const addProductCollectionToQueryPaginationParentOrAncestor = () => {
 };
 
 /**
- * Get the preview message for the Product Collection block based on the usesReference.
- * There are two scenarios:
- * 1. When usesReference is product, the preview message will be:
- * 	  "Actual products will vary depending on the product being viewed."
- * 2. For all other usesReference, the preview message will be:
- *    "Actual products will vary depending on the page being viewed."
- *
- * This message will be shown when the usesReference isn't available on the Editor side, but is available on the Frontend.
+ * Get the message to show in the preview label when the block is in preview mode based
+ * on the `usesReference` value.
  */
 export const getUsesReferencePreviewMessage = (
 	location: WooCommerceBlockLocation,
-	usesReference?: string[]
+	isUsesReferencePreviewMode: boolean
 ) => {
-	if ( ! ( Array.isArray( usesReference ) && usesReference.length > 0 ) ) {
-		return '';
+	if ( isUsesReferencePreviewMode ) {
+		if ( location.type === LocationType.Product ) {
+			return __(
+				'Actual products will vary depending on the product being viewed.',
+				'woocommerce'
+			);
+		}
+
+		return __(
+			'Actual products will vary depending on the page being viewed.',
+			'woocommerce'
+		);
+	}
+
+	return '';
+};
+
+export const getProductCollectionUIStateInEditor = ( {
+	location,
+	usesReference,
+	attributes,
+	hasInnerBlocks,
+}: {
+	location: WooCommerceBlockLocation;
+	usesReference?: string[] | undefined;
+	attributes: ProductCollectionAttributes;
+	hasInnerBlocks: boolean;
+} ): ProductCollectionUIStatesInEditor => {
+	const isInRequiredLocation = usesReference?.includes( location.type );
+	const isCollectionSelected = !! attributes.collection;
+
+	/**
+	 * Case 1: Product context picker
+	 */
+	const isProductContextRequired = usesReference?.includes( 'product' );
+	const isProductContextSelected =
+		( attributes.query?.productReference ?? null ) !== null;
+	if (
+		isProductContextRequired &&
+		! isInRequiredLocation &&
+		! isProductContextSelected &&
+		isCollectionSelected
```

Comment:
Nit: I know the order doesn't really matter in here but when I was thinking of this I was like "why we're checking if the collection is selected after verifying all of the conditions regarding collection first?"

What I'm trying to say is I'd reflect the actual order of conditions, because all three conditions doesn't matter if the collection is not selected:

```suggestion
                 isCollectionSelected && // is collection selected? Yes - Let's check some further details
		isProductContextRequired &&
		! isInRequiredLocation &&
		! isProductContextSelected
```